float64 colorX
float64 colorY

---

float64 valxx
float64 valyy